diff --git a/meson.build b/meson.build
index ddb74da..cd48bdf 100644
--- a/meson.build
+++ b/meson.build
@@ -205,7 +205,12 @@ endif
 
 if with_egl
    if cc.has_header('epoxy/egl.h', dependencies: epoxy_dep) and epoxy_dep.get_pkgconfig_variable('epoxy_has_egl') == '1'
-      gbm_dep = dependency('gbm', version: '>= ' + _gbm_ver, required: require_egl)
+      if with_minigbm_allocation
+         gbm_dep = dependency('minigbm', version: '>= ' + _gbm_ver, required: require_egl)
+      else
+         gbm_dep = dependency('gbm', version: '>= ' + _gbm_ver, required: require_egl)
+      endif
+
       have_egl = gbm_dep.found()
       if (have_egl)
          conf_data.set('HAVE_EPOXY_EGL_H', 1)
diff --git a/src/venus/vkr_device_memory.c b/src/venus/vkr_device_memory.c
index a014b74..b134194 100644
--- a/src/venus/vkr_device_memory.c
+++ b/src/venus/vkr_device_memory.c
@@ -5,7 +5,11 @@
 
 #include "vkr_device_memory.h"
 
+#ifdef ENABLE_MINIGBM_ALLOCATION
+#include <minigbm.h>
+#else
 #include <gbm.h>
+#endif
 
 #include "venus-protocol/vn_protocol_renderer_transport.h"
 
diff --git a/src/vrend_winsys_gbm.h b/src/vrend_winsys_gbm.h
index 84943fb..52ad135 100644
--- a/src/vrend_winsys_gbm.h
+++ b/src/vrend_winsys_gbm.h
@@ -25,7 +25,11 @@
 #ifndef VIRGL_GBM_H
 #define VIRGL_GBM_H
 
+#ifdef ENABLE_MINIGBM_ALLOCATION
+#include <minigbm.h>
+#else
 #include <gbm.h>
+#endif
 #include "vrend_iov.h"
 #include "virglrenderer.h"
 
